website/server/libs/applePayments.js

```diff
@@ -0,0 +1,172 @@
+import shared from '../../common';
+import iap from './inAppPurchases';
+import payments from './payments';
+import {
+  NotAuthorized,
+  BadRequest,
+} from './errors';
+import { model as IapPurchaseReceipt } from '../models/iapPurchaseReceipt';
+import {model as User } from '../models/user';
+import moment from 'moment';
+
+let api = {};
+
+api.constants = {
+  PAYMENT_METHOD_APPLE: 'Apple',
+  RESPONSE_INVALID_RECEIPT: 'INVALID_RECEIPT',
+  RESPONSE_ALREADY_USED: 'RECEIPT_ALREADY_USED',
+  RESPONSE_INVALID_ITEM: 'INVALID_ITEM_PURCHASED',
+  RESPONSE_STILL_VALID: 'SUBSCRIPTION_STILL_VALID',
+  RESPONSE_NO_ITEM_PURCHASED: 'NO_ITEM_PURCHASED',
+};
+
+api.verifyGemPurchase = async function verifyGemPurchase (user, receipt, headers) {
+  await iap.setup();
+  let appleRes = await iap.validate(iap.APPLE, receipt);
+  let isValidated = iap.isValidated(appleRes);
+  if (!isValidated) throw new NotAuthorized(api.constants.RESPONSE_INVALID_RECEIPT);
+  let purchaseDataList = iap.getPurchaseData(appleRes);
+  if (purchaseDataList.length === 0) throw new NotAuthorized(api.constants.RESPONSE_NO_ITEM_PURCHASED);
+  let correctReceipt = false;
+
+  // Purchasing one item at a time (processing of await(s) below is sequential not parallel)
+  for (let index in purchaseDataList) {
+    let purchaseData = purchaseDataList[index];
+    let token = purchaseData.transactionId;
+
+    let existingReceipt = await IapPurchaseReceipt.findOne({ // eslint-disable-line no-await-in-loop
```

Comment:
We've been using this await in a for loop since the gem purchases were added and it worked so far. During my testing now, I also didn't notice any issues. The reason for the eslint warning seems to rather be, that using await like this is not taking advantage of parallelization: http://eslint.org/docs/rules/no-await-in-loop